### PR TITLE
versioning: angular

### PR DESCRIPTION
**Pull request by timothy-fidelis**
- downgrade to angular 4
- fix all encountered errors with local run and packaging
- test in all higher angular versions (5 to 13)
- add npm script for building the library
- change the package version for publishing
- add prettier setup using tslint configurations
- install dev dependencies for linting and format
- add npm hooks on commit for fixing code standards according to config

### CODE REVIEW

**Comment by timothy-fidelis** [comment]
@nimlevy , as mentioned, created this pull request just for review and not merging.

File: .editorconfig [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Editor configuration, see https://editorconfig.org
+# Editor configuration, see http://editorconfig.org
```
Comment (nimlevy) [inline .editorconfig:1]: Is this needed?

**Review by nimlevy — APPROVED** [review]
Looks very configurable-based changes.
So I trust you

File: .editorconfig (hunk 1 above)
Comment (timothy-fidelis) [reply .editorconfig:1]: I actually did not do this changes, but just automatically created on `ng new <project>` using **@angular/cli version 4**

**Pull request merged**